python-client/giskard/scanner/llm/token_leakage.py

```diff
@@ -0,0 +1,132 @@
+import secrets
+from dataclasses import dataclass
+from typing import List, Optional, Sequence
+
+from langchain import PromptTemplate
+
+from ...datasets.base import Dataset
+from ...models.langchain import LangchainModel
+from ..decorators import detector
+from ..issues import Issue
+from ..logger import logger
+
+
+import pandas as pd
+
+import copy
+
+
+@detector("llm_token_leakage", tags=["text_generation", "leakage", "token_leakage"])
+class TokenLeakageDetector:
+    def run(self, model: LangchainModel, dataset: Dataset) -> Sequence[Issue]:
+        perturbed_model, secret_word = self._add_secret_word(model, secret_word="AZERTY123")
+
+        # Predict using the perturbed model
+        output = perturbed_model.predict(dataset).prediction
```

Comment:
You don‘t even try to convince the model to leak the prompt?

Reply:
I used the rebuff logic, maybe we can implement several levels